### ISSUE TITLE
`consult-theme` preview not working on graphical clients

### ISSUE THREAD

**Issue by aciceri** [minad/consult#785]
I remember this was working when I wasn't using daemon/clients and now it doesn't work anymore, I *guess* this is the cause but many other things changed in my config meanwhile so I'm not 100% confident.

I'm on Emacs 29 (pure GTK) using the daemon, when I run `consult-theme` in a graphical client I can move between candidates the theme preview seems to work only the first time I change candidate but giving this error:

```
Error in post-command-hook (consult--preview-post-command-hook): (error "Device <n> is not a termcap terminal device")
let*: Device <n> is not a termcap terminal device
```

Where `<n>` varies according to the number of active clients.
After this error I can still move between candidates but the preview doesn't work anymore.

If, even without restarting the daemon, I open a TTY client I can succesfully run `consult-theme` from there and it correctly changes theme also in the other (still opened) graphical clients.

**Comment by minad** [comment]
This sounds like a bug in your setup, pgtk or Emacs itself. Consult knows nothing about termcap devices. ;)

Can you please create a precise recipe for me to reproduce this? https://github.com/minad/consult#bug-reports

**Comment by aciceri** [comment]
Thanks, I've just found out what was responsible for this:

```
  (advice-add 'enable-theme
	      :after
	      #'(lambda (&rest rest)
		  (set-face-background 'vertical-border (face-background 'default))
		  (let* ((bg-color (face-attribute 'default :background))
			 (ansi-command (format "\033]11;#%s\007"
					       (string-remove-prefix "#" bg-color))))
```

I had to do so because when starting Emacs in TTY (not clients, just the `emacs` executable) the bottom line in terminals, which is almost always partially shown, doesn't change color according to the theme.
![image](https://user-images.githubusercontent.com/2318843/230588748-a82d5bd6-8e83-40f1-b9c7-73647542a721.png)
Probably I've to approach this problem differently now that I use the daemon.
Not related to `consult` though, just wanted to share my struggle.

Thanks for your fantastic job with this package!

**Comment by minad** [comment]
> Probably I've to approach this problem differently now that I use the
> daemon.
> Not related to |consult| though, just wanted to share my struggle.

My recommendation is to use a different workflow. You could use Emacs as
your main desktop application. Then you only need a single graphical
instance of Emacs and you also don't need the daemon. Emacs works very
well as a kind of supercharged terminal, where you can use the builtin
shell/eshell, file management/navigation, custom key bindings, etc. It
can also act as launcher for external applications.

> Thanks for your fantastic job with this package!

Thanks!

**Comment by aciceri** [comment]
I did something similar in the past (I've also used EXWM before which is the extreme of that I think) but now I prefer letting my windows manager (`sway`) manage windows. I've configured shortcuts that run `emacsclient` calling specifc functions, I use this to open terminals (which are Emacs frames with 1 `eat` or `eshell` buffer) or mail or capture an `org-roam` note for instance.
I wanted to try introducing `sway.el` in my workflow to control windows (emacs frames) from Emacs, I fear that to do so I need using the daemon. 

Also, right now I'm experimenting "remote development" on a server using ssh or `waypipe` for streaming the entire frame as a video, having a daemon always running is convenient because I can leave hundreds of buffer always open.
